### PR TITLE
Refactor simple widget editor config files

### PR DESCRIPTION
**Pull request by JacksonKearl**
In response to https://github.com/Microsoft/vscode/pull/54977#discussion_r204966401, where it was pointed out that all simple widget editors use the "DebugEditor" config file, which isn't very clear.

### CODE REVIEW

File: src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+				SelectionClipboard,
+				ContextMenuController,
+				SuggestController,
+				SnippetController2,
+				TabCompletionController,
+			]
+		};
+	}
+
+	public static getEditorOptions(style: 'editor' | 'htmlinput', ariaLabel?: string): IEditorOptions {
```
Comment (ramya-rao-a) [inline src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts]: I'd rather split this into 2 separate functions. How about `getEditorOptions` and `getEditorAsInputBoxOptions` ?

File: src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { SuggestController } from 'vs/editor/contrib/suggest/suggestController';
 import { SnippetController2 } from 'vs/editor/contrib/snippet/snippetController2';
 import { TabCompletionController } from 'vs/workbench/parts/snippets/electron-browser/tabCompletion';
 
-export class SimpleDebugEditor {
+export class SimpleWidgetEditorConfig {
```
Comment (isidorn) [inline src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts]: `SimpleEditorWidgetConfig` reads better to me

File: src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts [hunk 3]
```diff
@@ -54,4 +54,35 @@ export class SimpleDebugEditor {
 			}
 		};
 	}
+
+	public static getEditorAsInputBoxOptions(ariaLabel?: string): IEditorOptions {
```
Comment (isidorn) [inline src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts]: Why do we have both `getEditorOptions` and `getEditorAsInputBoxOptions`. The whole point of the simple editor widget is to be the input box so I do not understand the distinction.
If these options are true for all usages then just move them to `getEditorOptions`, if they are only used by extensions for some special use case then move them to extensions land.

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
@JacksonKearl thanks for the PR. Nice work to move this out of debug land.
I left some comments which I think should be addressed. Thanks

File: src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts (hunk 3 above)
Comment (JacksonKearl) [reply src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts]: I see it as we have two different types of SimpleEditors: those for code input and those for text input. Right now, the only text input is the Extensions Search, but if we were to add autosuggestions to settings or keybindings, that would be text as well.

**Review by ramya-rao-a — APPROVED** [review]

**Comment by isidorn** [comment]
@JacksonKearl nice work, I have merged it in
I have pushed 86c743c28e which I believe simplifies some naming.

**Pull request merged**